assembly/src/todo/instruction/mod.rs

```diff
@@ -0,0 +1,238 @@
+mod field_ops;
+mod flow_ops;
+mod io_ops;
+
+use vm_core::{code_blocks::CodeBlock, Operation};
+
+use crate::{parsers::Instruction, AssemblerError};
+
+use super::AssemblerContext;
+
+impl AssemblerContext {
+    pub(super) fn compile_instruction(
+        &mut self,
+        blocks: &mut Vec<CodeBlock>,
+        span: &mut Vec<Operation>,
+        instruction: &Instruction,
+    ) -> Result<(), AssemblerError> {
```

Comment:
As I mentioned in one of previous comments, I would change the signature to be:
```Rust
pub(super) fn compile_instruction(
    &self,
    instruction: &Instruction,
    span: &mut Vec<Operation>,
) -> Result<Option<CodeBlock>, AssemblerError>
```
Note that `self` here should be `Assembler` and it shouldn't be mutable.

Reply:
fn refactored